charts/seed-monitoring/charts/core/charts/prometheus/rules/node-exporter.rules.yaml

```diff
@@ -49,6 +49,30 @@ groups:
       summary: Node is under disk pressure.
       description: Node {{ $labels.node }} is under disk pressure
 
+  - alert: PodsRequestCPUApproachNodeCapacity
+    expr: sum by (node)(kube_pod_container_resource_requests_cpu_cores) / sum by (node)(kube_node_status_allocatable{resource="cpu"}) > 80
```

Comment:
Unfortunately I think there are some challenges with the approach of this PR.

A general comment first: have you managed to trigger such an alert in a local / dev setup?
My concern is that if the sum of the CPU requests of the pods on a given node exceeds 80 % of the node's allocatable CPU capacity (this is what the promql expression checks if we were to fix the off by a factor of 100 mistake: instead of 80, you'd need 0.8 to express 80%), it means that the node is well utilized, maybe with multiple small pods, which is perfectly fine. The problem I see is that actually it does not check the condition that is the intent of this PR: if a single pod is scaled up by the VPA beyond the allocatable capacity of the node and hence the pod can't even be scheduled. So you would have to check if a single pod's request is approaching the node's allocatable capacity. On the other hand, if it is only "approaching" the capacity but the pod can still be scheduled, it is fine - no need to alert. It becomes critical when the pod can not be scheduled because the request exceeds the allocatable capacity. Note that the scale up could "jump" over the range of 80-100%, e.g. from 65% to 105%, so that we might not catch it at all with this alert. Furthermore, if the pod is not scheduled (stuck in pending state), the kube_pod_container_resource_requests_cpu_cores metric will not be reported for a node, and hence the promql expression will not work.

Another issue I see is that in the control plane Prometheus (where the alert is added), the kube-state-metrics metrics (like kube_pod_container_resource_requests_cpu_cores) are scraped for the

- kube-system namespace of the shoot cluster, and only for the kube-system namespace (excluding the user workload)
- for the control plane namespace in the seed cluster, and the metrics are "mapped" to the kube-system namespace

Adding up these requests is not meaningful (some of the pods run in the shoot, some in the seed) and comparing them with the shoot node's capacity which also executes user workload is not meaningful either.

So I think we need to consider this or even alternative approaches to reach the goal of this PR. Generally, trying to "trigger" the alert by manually scaling up a component to request more CPU than the node's allocatable capacity and verifying that the alert actually fires is the only way I'm aware of to efficiently get feedback on any specific approach and to make sure to avoid adding alerts that can never fire.

An alternative approach could be to check the symptom of scaling up "too high": the pod will not be scheduled to any node and will stay in the pending state forever. (Note that this approach would also need to be confirmed in a dev setup.) Another idea is to check a similar condition in the "cache" Prometheus (also known as just "prometheus" in the garden namespace, see https://gardener.cloud/docs/gardener/monitoring/) because it gets the kube-state-metrics metrics only for the seed (not for the shoot and the seed) and for all the pods in the seed (not limited to the kube-system namespace.) Note that the metric `kube_pod_container_resource_requests_cpu_cores` has been renamed with the recent ksm upgrade (see https://github.com/gardener/gardener/pull/6224/commits/21fcb06f8307ffa41b073e0d6265b46fc8877b26) and the form `kube_pod_container_resource_requests{resource="cpu", unit="core"}` should be used instead. Furthermore, it seems that these metrics are also reported for the pods in the `Completed` and `ContainerStatusUnknown` states which resulted in not really meaningful values when I added them up in a test environment. So I'm not sure if the approach of this PR can be "fixed" easily. I suggest triggering the specific situation first in a dev setup and exploring the various Prometheus instances and metrics to capture that condition.